src/validation.cpp

```diff
@@ -2601,6 +2601,7 @@ CBlockIndex* AddToBlockIndex(const CBlockHeader& block)
         pindexNew->nHeight = pindexNew->pprev->nHeight + 1;
         pindexNew->BuildSkip();
     }
+    pindexNew->nTimeMax = (pindexNew->pprev ? std::max(pindexNew->pprev->nTimeMax, pindexNew->nTime) : pindexNew->nTime);
```

Comment:
How does nTimeMax get assigned when the block index is loaded on startup? Do you need to add a similar assignment to `LoadBlockIndexDB`?